### PR TITLE
Disable SDPA FlashAttention backward on sm86 to sm89 inclusive - Test on SM89

### PR DESCRIPTION
**Pull request by cpuhrsch**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: aten/src/ATen/native/transformers/cuda/sdp_utils.h [hunk 1]
```diff
@@ -533,12 +533,22 @@ inline bool check_gpu_sm86_head_dim_128(sdp_params params, bool debug) {
...
+    bool debug) {
+  // Flash Attention will raise an error in the backward pass if the head_dim
+  // size is 128 And the device is not sm80, the other head_dim check catches
+  // everything but sm86
+  auto dprops = at::cuda::getCurrentDeviceProperties();
+  bool is_sm86_or_newer = (dprops->major == 8) && (dprops->minor >= 6);
+  bool is_sm90_or_newer = (dprops->major == 9) && (dprops->minor >= 0);
+  if (!check_requires_grad(params, false) &&
+      !check_gpu_sm86_head_dim_128(params, false) &&
+      !is_sm86_or_newer &&
```
Comment (ngimel) [inline aten/src/ATen/native/transformers/cuda/sdp_utils.h:547]: `is_sm86_or_newer` will evaluate to `true` for `sm_89`, thus this line will evaluate to `false`

File: aten/src/ATen/native/transformers/cuda/sdp_utils.h [hunk 2]
```diff
@@ -533,12 +533,22 @@ inline bool check_gpu_sm86_head_dim_128(sdp_params params, bool debug) {
...
-      TORCH_WARN("Flash attention currently doesn't support training with head_dim == 128 on sm86.");
+inline bool check_requires_grad_and_head_dim_128_and_sm86_or_greater(
+    sdp_params params,
+    bool debug) {
+  // Flash Attention will raise an error in the backward pass if the head_dim
+  // size is 128 And the device is not sm80, the other head_dim check catches
+  // everything but sm86
+  auto dprops = at::cuda::getCurrentDeviceProperties();
+  bool is_sm86_or_newer = (dprops->major == 8) && (dprops->minor >= 6);
+  bool is_sm90_or_newer = (dprops->major == 9) && (dprops->minor >= 0);
```
Comment (malfet) [inline aten/src/ATen/native/transformers/cuda/sdp_utils.h:544]: Why do you need separate `sm90_or_newer`?
```suggestion
  bool is_sm86_or_newer = ((dprops->major == 8) && (dprops->minor >= 6)) || (dprops->major > 8);
```

Comment (ngimel) [reply aten/src/ATen/native/transformers/cuda/sdp_utils.h:544]: sm_90 should work according to checks in backward (but we can't test)

File: aten/src/ATen/native/transformers/cuda/sdp_utils.h [hunk 3]
```diff
@@ -533,12 +533,18 @@ inline bool check_gpu_sm86_head_dim_128(sdp_params params, bool debug) {
...
-      TORCH_WARN("Flash attention currently doesn't support training with head_dim == 128 on sm86.");
+inline bool check_requires_grad_and_head_dim_gt64_and_sm_gt80(
+    sdp_params params,
+    bool debug) {
+  // Flash Attention will raise an error in the backward pass if the head_dim
+  // size is 128 And the device is not sm80, the other head_dim check catches
+  // everything but sm86
+  auto dprops = at::cuda::getCurrentDeviceProperties();
+  bool is_sm86_or_newer = ((dprops->major == 8) && (dprops->minor > 0)) || (dprops->major > 8);
+  if (is_sm86_or_newer && (params.query.sym_size(-1) > 64)) {
```
Comment (ngimel) [inline aten/src/ATen/native/transformers/cuda/sdp_utils.h:544]: you still need to check if requires grad?

**Pull request closed without merging**